### ISSUE TITLE
Broken notification policies after enabling unified alerting

### ISSUE THREAD

**Issue by penekk** [grafana/grafana#46888]
**What happened**:

Enabling unified alerting triggered migration during which a unique `rule_uid` was added for each end every alert rule. Automatically generated notification policies try to match those alert rules by `rule_uid` label but the value is wrapped with quotes, which breaks the entire chain, thus not sending any notifications for migrated alerts.

Edit: looks like those quotes are simply escaped when saved to database. I think migration process should not wrap notification policy matchers.

**What you expected to happen**:

Notification policies are properly matching automatically generated alert rules.

**How to reproduce it (as minimally and precisely as possible)**:

Trigger migration by enabling unified alerting in grafana config and starting the server.
Create notification policy matching `rule_uid` value wrapped in quotes (or use a migrated one)
![asd](https://user-images.githubusercontent.com/12582893/159727615-1dd36d71-a748-4f6f-96d0-ace768d55a86.png)
Confirm that no alert is matched by notification policy and therefore no message is sent to a contact point of your choice.

**Anything else we need to know?**:

We can, and probably will, edit those manually and fix our policies, but I wonder if that's actually a bug or something we might have overlooked/misconfigured.

**Environment**:
- Grafana version: 8.4.1
- Data source type & version: irrelevant
- OS Grafana is installed on: centos 7.8.2003
- User OS & Browser: irrelevant?
- Grafana plugins: irrelevant?
- Others:

**Comment by zuchka** [comment]
> Trigger migration by enabling unified alerting in grafana config and starting the server.

can you add more details for reliably reproducing this? 🙇 

- also please list all configuration options related to alerting. You can find these in the `Alerting` and `Unified Alerting` sections of Grafana's config file. If you are now using or have previously used the beta version of `ngalert` (released with Grafana 8), please note that too.
  - you can use [this table](https://github.com/grafana/grafana/pull/42200#issue-1062269188) to better understand how configuration options can interact with each other. It also has some useful info about migrations.
- Also, can you please search for [open issues in this repo](https://github.com/grafana/grafana/issues) that sound similar to your problem?

**Comment by penekk** [comment]
> can you add more details for reliably reproducing this? :bow:

> * also please list all configuration options related to alerting. You can find these in the `Alerting` and `Unified Alerting` sections of Grafana's config file. If you are now using or have previously used the beta version of `ngalert` (released with Grafana 8), please note that too.

Sure, I assumed I was explicit enough - allow me to go full verbose mode ;)
Grafana version is 8.4.1 and those are instances that were upgraded from previous versions, multiple times over the years, which in this case means:

* no `[alerting]` or `[unified alerting]` options were explicitly set prior to migration - we were fine using Legacy alerting until we were confident that Unified alerting is solid enough to migrate
* for the migration process we stopped the instances, made changes to configs (below) and brought instances back up so the application can perform migration (teammate performing the actual task unfortunately brought them both up, but due to transactional nature of this we don't think that is a problem either).

```
[alerting]
enabled = false
[unified alerting]
enabled = true
execute_alerts = true
```
(later on we added ha_peers which we forgot about, but that is rather irrelevant here too)


No beta version of new alerting was used, consciously, I've been following this subject for a long time now.
>  you can use [this table](https://github.com/grafana/grafana/pull/42200#issue-1062269188) to better understand how configuration options can interact with each other. It also has some useful info about migrations.

I think we can understand combinations of these options pretty well. The problem wasn't unexpected behaviour of any of these. The problem is with values used for notification policies during what seems a proper, successful migration.
> Also, can you please search for [open issues in this repo](https://github.com/grafana/grafana/issues) that sound similar to your problem?

I only found https://github.com/grafana/grafana/issues/45443 this issue which although aligns with my experience to some degree :) isn't exactly the same thing.

I hope this gives you more clarity.
